TestingHomework.Tests/ProductAdminManagerTests.cs

```diff
@@ -1,108 +1,146 @@
-﻿using System;
+﻿using Accessors.DatabaseAccessors;
+using DeepEqual.Syntax;
+using Managers.LazyCollectionOfAllManagers;
+using Ninject;
+using NUnit.Framework;
+using Shared.DatabaseContext;
+using Shared.DatabaseContext.DBOs;
+using Shared.DataContracts;
+using Shared.DependencyInjectionKernel;
+using System;
 using System.Collections.Generic;
-using System.Text;
-using Xunit;
-using TestingHomework_Discounts.Managers;
-using TestingHomework_Discounts;
 using System.Linq;
+using Telerik.JustMock.AutoMock;
+using Test.NUnitExtensions;
+using Tests.DataPrep;
+using Tests.ManagerTests;
 
 namespace TestingHomework.Tests
 {
-    public class ProductAdminManagerTests
+    //[TestFixture_Prefixed(typeof(ProductAdminManager), false)]
+    [TestFixture_Prefixed(typeof(ProductAdminManager), true)]
+    public class ProductAdminManagerTests : ManagerTestBase
     {
-        ProductAdminManager manager = new ProductAdminManager();
+        ProductAdminManager manager;
+        MockingContainer<ProductAdminManager> mockContainer = new MockingContainer<ProductAdminManager>();
 
-        [Fact]
-        public void GetProduct_NoProduct()
+        public ProductAdminManagerTests() : this(false) { }
+        public ProductAdminManagerTests(bool isIntegration = false)
         {
+            // this constructor allows for integration test reuse
+            if (isIntegration)
+            {
+                // Implicit self binding allows us to get a concrete class with fulfilled dependencies
+                // https://github.com/ninject/ninject/wiki/dependency-injection-with-ninject
+                Ninject.IKernel kernel = DependencyInjectionLoader.BuildKernel();
+                manager = kernel.Get<ProductAdminManager>();
+                dataPrep = new DiscountsDataPrep(true);
+            }
+            else
+            {
+                manager = mockContainer.Instance;
+                // dataPrep non-persistant by default in base class
+            }
+        }
 
-            var actualEmpty = manager.GetProduct(Guid.Empty);
-            Assert.Null(actualEmpty);
+        public override void OnCleanup()
+        {
+            using (TestingHomeworkContext db = new TestingHomeworkContext())
```

Comment:
Why is this needed? If your transactions work properly, then test data should be wiped from the database at the end of each test by discarding the transaction. 

If there is data being left in the DB I would encourage you to understand why it isn't part of the transaction rather than clean it up separately